thirdparty/faiss/faiss/IndexFlat.cpp

```diff
@@ -354,6 +354,7 @@ struct FlatL2WithNormsDis : FlatCodesDistanceComputer {
     void set_query(const float* x) override {
         q = x;
         query_l2norm = fvec_norm_L2sqr(q, d);
+        query_l2norm = (query_l2norm == 0.0 ? 1.0 : query_l2norm);
```

Comment:
no, because it uses norm for distance = x^2 - 2xy + y^2 calculation, not for a cosine-related code